portfolio/portfolio.go

```diff
@@ -508,18 +503,30 @@ func (b *Base) GetPortfolioAddressesGroupedByCoin() map[currency.Code][]string {
 }
 
 // StartPortfolioWatcher observes the portfolio object
-func (b *Base) StartPortfolioWatcher(ctx context.Context, sleepDelay time.Duration) error {
+func (b *Base) StartPortfolioWatcher(ctx context.Context, interval time.Duration) error {
 	if len(b.Addresses) == 0 {
 		return errNoPortfolioItemsToWatch
 	}
 
-	if sleepDelay <= 0 {
-		sleepDelay = time.Minute * 10
+	if interval <= 0 {
+		interval = defaultInterval
 	}
 
 	log.Infof(log.PortfolioMgr, "PortfolioWatcher started: Have %d entries in portfolio.\n", len(b.Addresses))
 
-	ticker := time.NewTicker(sleepDelay)
+	updatePortfolio := func() {
+		for key, value := range b.GetPortfolioAddressesGroupedByCoin() {
+			if err := b.UpdatePortfolio(ctx, value, key); err != nil {
+				log.Errorf(log.PortfolioMgr, "PortfolioWatcher: UpdatePortfolio error: %s for currency %s", err, key)
+				continue
+			}
+			log.Debugf(log.PortfolioMgr, "PortfolioWatcher: Successfully updated address balance for %s", key)
+		}
+	}
```

Comment:
🐒 
This func is not closuring any vars apart from `b`.
So I think it should be a private method on Base instead.

🥃  Just processed what `Base` is here. Bit of an overloaded type name; We should probably drive by that at some point.